firestore.rules

```diff
@@ -338,7 +347,14 @@ service cloud.firestore {
         function isValidCommentUpdateRequest() {
           return userIsRequestUser() && preservesReadOnlyCommentFields();
         }
-
+        
+        function userOwnsParentHistoryDocument() {
+          return getHistoryParentDocumentOwner() == string(request.auth.token.platform_user_id);
+        }
+        function teacherCanReadStudentHistory() {
+          return getHistoryDocumentContext() == string(request.auth.token.class_hash);
```

Comment:
I think this is what we talked about, though I notice that the comments are looking to see whether the user is in the list of teachers so I'm not confident I got this right.

Reply:
I switched this over to using `teacherCanAccessDocumentand` removed `teacherCanReadStudentHistory`.